### PR TITLE
improved functionality

### PR DESCRIPTION
**Pull request by Walle10-0**
wanted to do a pull to main before we make more changes to dev

### CODE REVIEW

**Review by Walle10-0 — COMMENTED** [review]
changes :
 - lombok added
 - files moved
 - unnecessary build/compile stuff removed
 - package rename from "coe.dataCollection" to "coe.datacollection"
 - added and incorporated service class for user
 - added and incorporated user DTO
 - removed "users" from Department class
 - finding users by department uses user repository, not department repository
 - indentation fixes
 - MainController instead of TestController
 - /api instead of /demo

**Comment by Walle10-0** [comment]
I need somebody else to take a quick glance at the changes so they can be approved

**Review by Walle10-0 — COMMENTED** [review]
added GUI section + finished DTO

File: backend/pom.xml [hunk 1]
```diff
@@ -9,12 +9,12 @@
...
 	<groupId>coe</groupId>
-	<artifactId>dataCollection</artifactId>
+	<artifactId>datacollection</artifactId>
 	<version>0.0.1-SNAPSHOT</version>
-	<name>dataCollection</name>
+	<name>datacollection</name>
 	<description>Demo project for Spring Boot</description>
 	<properties>
-		<java.version>21</java.version>
+		<java.version>17</java.version>
```
Comment (Diamondragon4) [inline backend/pom.xml]: You switched Java versions by accident

**Review by Diamondragon4 — APPROVED** [review]

**Pull request merged**